frame/staking-rewards/src/lib.rs

```diff
@@ -363,7 +382,87 @@ pub mod pallet {
 			};
 			let new_position = StakeCount::<T>::increment()?;
 			Stakes::<T>::insert(new_position, new_stake);
+			Self::deposit_event(Event::<T>::SplitPosition {
+				positions: vec![*position, new_position],
+			});
 			Ok([*position, new_position])
 		}
 	}
+
+	impl<T: Config> ProtocolStaking for Pallet<T> {
+		type AssetId = T::AssetId;
+		type AccountId = T::AccountId;
+		type RewardPoolId = T::RewardPoolId;
+		type Balance = T::Balance;
+
+		fn accumulate_reward(
+			_pool: &Self::RewardPoolId,
+			_reward_currency: Self::AssetId,
+			_reward_increment: Self::Balance,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		#[transactional]
+		fn transfer_reward(
+			from: &Self::AccountId,
+			pool: &Self::RewardPoolId,
+			reward_currency: Self::AssetId,
+			reward_increment: Self::Balance,
+		) -> DispatchResult {
+			RewardPools::<T>::try_mutate(pool, |reward_pool| {
+				match reward_pool {
+					Some(reward_pool) => {
+						match reward_pool.rewards.get_mut(&reward_currency) {
+							Some(mut reward) => {
+								let new_total_reward = reward.total_rewards + reward_increment;
+								ensure!(
+									(new_total_reward - reward.total_dilution_adjustment) <=
+										reward.max_rewards,
+									Error::<T>::MaxRewardLimitReached
+								);
+								reward.total_rewards = new_total_reward;
+								let pool_account = Self::account_id(pool);
+								T::Assets::transfer(
+									reward_currency,
+									from,
+									&pool_account,
+									reward_increment,
+									false,
+								)?;
+							},
+							None => {
+								// new reward asset so only pool owner is allowd to add.
+								ensure!(
+									*from == reward_pool.owner,
+									Error::<T>::OnlyPoolOwnerCanAddNewReward
+								);
+								let reward = Reward {
+									asset_id: reward_currency,
+									total_rewards: reward_increment,
+									total_dilution_adjustment: T::Balance::zero(),
+									max_rewards: reward_increment,
```

Comment:
Let's set some default value that say like 1M units here so that it won't stop accepting the transfers the immediately